webapp/src/Components/Replay/ReplayViewer/ReplayViewer.tsx

```diff
@@ -33,15 +39,48 @@ export class ReplayViewer extends React.PureComponent<Props, State> {
                     </Typography>
                 </Grid>
                 <Grid item xs={12}>
-                    {this.state.replayData &&
-                    <ThreeScene replayData={this.state.replayData}/>}
+                    {
+                        !this.state.replayData &&
+                        <Typography>Loading...</Typography>
+                    }
+                    {
+                        this.state.replayData &&
+                        <ThreeScene replayData={this.state.replayData} frame={this.state.currentFrame}/>
+                    }
+                </Grid>
+                <Grid item xs={6}>
+                    <label>Frame:</label>
+                    <input type="number" value={this.state.currentFrame} onChange={this.setCurrentFrame} />
+                </Grid>
+                <Grid item xs={6}>
+                    <Typography>
+                        Ball Position:
+                        {
+                            this.state.replayData &&
+                            this.state.replayData.ball[this.state.currentFrame][0]
+                        },
+                        {
+                            this.state.replayData &&
+                            this.state.replayData.ball[this.state.currentFrame][1]
+                        },
+                        {
+                            this.state.replayData &&
+                            this.state.replayData.ball[this.state.currentFrame][2]
+                        }
+                    </Typography>
                 </Grid>
             </Grid>
         )
     }
 
-    public getReplayPositions = () => {
-        getReplayViewerData(this.props.match.params.id)
-            .then((data: any) => this.setState({replayData: data.json()}))
+    public getReplayPositions = async() => {
+        const data: any = await getReplayViewerData(this.props.replay.id)
```

Comment:
Is adding async+await everywhere better than doing `getData(id).then((data) => this.setState({data}))`?

Reply:
It's functionally the same thing, but the syntax (IMO) is easier to read, especially if you are doing more than 1 operation in the `.then()`.